### PR TITLE
Spelling fixes for codespell

### PR DESCRIPTION
**Pull request by derobins**
Unclear why codespell missed these earlier, but here's the fixes.

### CODE REVIEW

File: test/cache.c [hunk 1]
```diff
@@ -9853,15 +9853,6 @@ check_flush_cache__flush_op_eviction_test(H5F_t *file_ptr)
  * Programmer:    John Mainzer
  *              1/12/05
  *
- * Modifications:
```
Comment (fortnern) [inline test/cache.c:9856]: Why were these lines removed?

**Review by jhendersonHDF — APPROVED** [review]

File: test/cache.c (hunk 1 above)
Comment (derobins) [reply test/cache.c:9856]: Modification lines in general are going away. It was easier to delete them than correct them.

Comment (bmribler) [reply test/cache.c:9856]: I know you've been doing that and I heard your reasoning.  It's just sometime I found some good explanation in these Modifications notes very helpful.  I'm not pushing to keep them though...
